c10/core/DispatchKeySet.cpp

```diff
@@ -103,6 +104,15 @@ DispatchKeySet getAutocastRelatedKeySetFromBackend(DispatchKey t) {
     case DispatchKey::CUDA:
     case DispatchKey::XLA:
       return DispatchKeySet(DispatchKey::AutocastCUDA);
+    case DispatchKey::Lazy:
```

Comment:
I don't think we can fix this properly until we redesign lazy tensor as a 'mode'.  That said, you could work around for now by checking `LTC_TS_CUDA` env here.